x-pack/plugins/observability_onboarding/server/routes/elastic_agent/generate_yml.ts

```diff
@@ -25,6 +27,16 @@ export const generateYml = ({
   logfileId: string;
 }) => {
   const customConfigYaml = load(customConfigurations ?? '');
+  const processors = [
+    {
+      add_fields: {
+        target: 'service',
+        fields: {
+          name: serviceName,
+        },
+      },
+    },
+  ];
```

Comment:
I'd love to have a test that produces this yaml file, and validates it with elastic agent somehow.

Reply:
yes, I'll add the tests in the context of https://github.com/elastic/kibana/issues/159451.